src/handler/mod.rs

```diff
@@ -0,0 +1,145 @@
+/// This module most of low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Error, Result};
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
+    if !force {
+        match get_boot_counter() {
+            Some(t) if t <= 0 => bail!("countdown ended, check greenboot-rollback status"),
+            None => bail!("boot_counter is not set"),
+            _ => {}
```

Comment:
`_ => {}` is due to the condition check `if ` logic in line no. 13  that is excluding the the positive integers, so  if `{}` looks bad or not the best practice it wont hurt pulling : 
```
log::info!("restarting system");
                let status = Command::new("systemctl").arg("reboot").status()?;
                if status.success() {
                    return Ok(());
                }
```
which is there below outside `match`

Reply:
pondering more on this: if `boot_counter is not set` then counter as `None` seems meaningful to me.
also I understand this may loose some of the simplicity of using:
`if let Some(current_counter) = get_boot_counter()` thats has been used extensively.